miditok/tokenizations/remi_plus.py

```diff
@@ -0,0 +1,814 @@
+                )
+            )
+            duration = note.end - note.start
+            index = np.argmin(np.abs(dur_bins - duration))
+            events.append(
+                Event(
+                    type="Duration",
+                    value=".".join(map(str, self.durations[index])),
+                    time=note.start,
+                    desc=f"{duration} ticks",
+                )
+            )
+            previous_note_end = max(previous_note_end, note.end)
+
+        events.sort(key=lambda x: (x.time, self._order(x)))
+        return events
+
+    @_out_as_complete_seq
+    def track_to_tokens(self, track: Instrument) -> TokSequence:
+        r"""Converts a track (miditoolkit.Instrument object) into a sequence of tokens (:class:`miditok.TokSequence`).
+
+        NOTE: REMI+ is REMI-based extended representation for multi-track encodings in single sequence. Then if you'd
+        like to get only single-track tokens, use REMI.
+
+        :param track: MIDI track to convert
+        :return: :class:`miditok.TokSequence` of corresponding tokens.
+        """
+        events = self.__notes_to_events(
+            [(n, (track.program, track.is_drum)) for n in track.notes]
+        )
+        return TokSequence(events=events)  # type: ignore
+
+    def tokens_to_track(
+        self,
+        tokens: Union[TokSequence, List, np.ndarray, Any],
+        time_division: Optional[int] = TIME_DIVISION,
+        program: Optional[Tuple[int, bool]] = (0, False),
+    ) -> None:
+        r"""NOT RELEVANT / IMPLEMENTED FOR REMIPlus
+        Use tokens_to_midi instead
+
+        :param tokens: sequence of tokens to convert. Can be either a Tensor (PyTorch and Tensorflow are supported),
+                a numpy array, a Python list or a TokSequence.
+        :param time_division: MIDI time division / resolution, in ticks/beat (of the MIDI to create)
+        :param program: the MIDI program of the produced track and if it drum, (default (0, False), piano)
+        :return: None
+        """
+        pass
+
+    def _midi_to_tokens(self, midi: MidiFile, *args, **kwargs) -> TokSequence:
+        r"""Converts a preprocessed MIDI object to a sequence of tokens.
+
+        :param midi: the MIDI objet to convert.
+        :return: sequences of tokens.
+        """
+        # Convert each track to tokens
+        notes_with_program = [
+            ((note, (track.program, track.is_drum)))
+            for track in midi.instruments
+            for note in track.notes
+        ]
+        notes_with_program.sort(key=lambda n: (n[0].start, n[0].pitch))
+        events = self.__notes_to_events(notes_with_program)
+        tok_sequence = TokSequence(events=cast(List[Union[Event, List[Event]]], events))
+        self.complete_sequence(tok_sequence)
+        return tok_sequence
+
+    @_in_as_seq()
+    def tokens_to_midi(
+        self,
+        tokens: Union[TokSequence, List, np.ndarray, Any],
+        _=None,
```

Comment:
Maybe adding a note in the docstring ?
In Octuple / MuMIDI there is: `:param _: unused, to match parent method signature`